docs/tutorials/zarr-v3.md

```diff
@@ -0,0 +1,101 @@
+
+```python
+import zarr
+import anndata as ad
+from collections.abc import Mapping
+from typing import Any
+
+ad.settings.zarr_write_format = 3 # Absolutely crucial! Sharding is only for the v3 file format!
+
+def write_sharded(group: zarr.Group, adata: ad.AnnData):
+    def callback(
+        func: ad.experimental.Write,
+        g: zarr.Group,
+        k: str,
+        elem: ad.typing.RWAble,
+        dataset_kwargs: Mapping[str, Any],
+        iospec: ad.experimental.IOSpec,
+    ):
+        if iospec.encoding_type in {"array"}:
+            dataset_kwargs = {
+                "shards": tuple(int(2 ** (16 / len(elem.shape))) for _ in elem.shape),
+                **dataset_kwargs,
+            }
+            dataset_kwargs["chunks"] = tuple(i // 2 for i in dataset_kwargs["shards"])
+        elif iospec.encoding_type in {"csr_matrix", "csc_matrix"}:
+            dataset_kwargs = {"shards": (2**16,), "chunks": (2**8,), **dataset_kwargs}
+        func(g, k, elem, dataset_kwargs=dataset_kwargs)
+
+    return ad.experimental.write_dispatched(group, "/", adata, callback=callback)
+```
+
+However, `zarr-python` can be slow with sharding throughput as well as writing throughput.
+Thus if you wish to speed up either writing, sharding, or both (or receive a modest speed-boost for reading), a bridge to the `zarr` implementation in Rust {doc}`zarrs-python <zarrs:index>` can help with that (see the [zarr-benchmarks]):
+
+```
+uv pip install zarrs
+```
+
+```python
+import zarr
+import zarrs
+zarr.config.set({"codec_pipeline.path": "zarrs.ZarrsCodecPipeline"})
+```
+
+However, this pipeline is not compatible with all types of zarr store, especially remote stores and there are limitations on where rust can give a performance boost for indexing.
+We therefore recommend this pipeline for writing full datasets and reading contiguous regions of said written data.
+
+## Codecs
+
+The default `zarr-python` v3 codec for the v3 format is no longer `blosc` but `zstd`.
+While `zstd` is more widespread, you may find its performance to not meet your old expectations.
+Therefore, we recommend passing in the {class}`zarr.codecs.BloscCodec` to `compressor` on {func}`~anndata.AnnData.write_zarr` if you wish to return to the old behavior.
+
+There is currently a bug with `numcodecs` that prevents data written from other non-numcodecs `zstd` implementations from being read in by the default zarr pipeline (to which the above rust pipeline falls back if it cannot handle a datatype or indexing scheme, like `vlen-string`): {issue}`zarr-developers/numcodecs#424`.
+Thus is may be advisable to use `BloscCodec` with `zarr` v3 file format data if you wish to use the rust-accelerated pipeline until this issue is resolved.
+
+The same issue with `zstd` applies to data that may eventually be written by the GPU `zstd` implementation (see below).
+
+## GPU i/o
+
+At the moment, it is unlikely your `anndata` i/o will work if you use `zarr.enable_gpu <zarr:user-guide-gpu>`.
```

Comment:
Isn’t it `zarr.config.enable_gpu`? Maybe just link to the document? The header is “Using GPUs with Zarr”, which would fit into the text if you want.

Reply:
I did not see that in the `objects.inv` when I searched it, but maybe I missed it

Reply:
Yes, sorry, but not sure what you mean by "just link to the document"